File: tests/solutions_js/binary_tree_level_order_traversal.js
class Solution {
    levelOrder(root) {
        if (root === null) return [];
        const result = [];
        const queue = [root];
        while (queue.length > 0) {
            const levelSize = queue.length;
            const level = [];
            for (let i = 0; i < levelSize; i++) {
                const node = queue.shift();
                level.push(node.val);
                if (node.left) queue.push(node.left);
                if (node.right) queue.push(node.right);
            }
            result.push(level);
        }
        return result;
    }
}

File: tests/solutions_cpp/binary_tree_level_order_traversal.cpp
#include "json_helper.h"

class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        if (!root) return result;
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()) {
            int sz = q.size();
            vector<int> level;
            for (int i = 0; i < sz; i++) {
                TreeNode* n = q.front(); q.pop();
                level.push_back(n->val);
                if (n->left) q.push(n->left);
                if (n->right) q.push(n->right);
            }
            result.push_back(level);
        }
        return result;
    }
};

int main() {
    auto args = parse_args();
    Solution sol;
    auto root = build_tree(args[0]);
    auto result = sol.levelOrder(root);
    cout << to_json(result) << endl;
    return 0;
}
